pkg/store/bucket.go

```diff
@@ -786,29 +792,20 @@ func blockSeries(
 		return newBucketSeriesSet(res), indexr.stats, nil
 	}
 
-	// Preload all chunks that were marked in the previous stage.
-	if err := chunkr.preload(); err != nil {
+	if err := chunkr.preload(res, req.Aggregates); err != nil {
 		return nil, nil, errors.Wrap(err, "preload chunks")
 	}
 
-	// Transform all chunks into the response format.
-	for _, s := range res {
-		for i, ref := range s.refs {
-			chk, err := chunkr.Chunk(ref)
-			if err != nil {
-				return nil, nil, errors.Wrap(err, "get chunk")
-			}
-			if err := populateChunk(&s.chks[i], chk, req.Aggregates); err != nil {
-				return nil, nil, errors.Wrap(err, "populate chunk")
-			}
-		}
-	}
 	return newBucketSeriesSet(res), indexr.stats.merge(chunkr.stats), nil
 }
 
-func populateChunk(out *storepb.AggrChunk, in chunkenc.Chunk, aggrs []storepb.Aggr) error {
+func populateChunk(out *storepb.AggrChunk, in chunkenc.Chunk, aggrs []storepb.Aggr, savior func([]byte) ([]byte, error)) error {
```

Comment:
What `savior` means in this context? (: Why such name?

Reply:
There was no religious reference - this function rescues byte slice from the loop iteration, preserving its contents in internal buffer, acts as a saviour (from a byte slice perspective).

It probably was coffee overdose. Renamed to plain `save` to avoid those sidelong glances :)